xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Memo.java

```diff
@@ -37,24 +48,66 @@
    * @return An {@link Optional} of type {@link String} containing the memo data.
    */
   @JsonProperty("MemoData")
-  Optional<String> memoData();
+  public abstract Optional<String> memoData();
 
   /**
-   * Hex value representing characters allowed in URLs. Conventionally containing information on how the memo
-   * is encoded.
+   * Hex value representing characters allowed in URLs. Conventionally containing information on how the memo is
+   * encoded.
    *
    * @return An {@link Optional} of type {@link String} containing the memo format.
    */
   @JsonProperty("MemoFormat")
-  Optional<String> memoFormat();
+  public abstract Optional<String> memoFormat();
 
   /**
-   * Hex value representing characters allowed in URLs. Conventionally, a unique relation that defines the format
-   * of this memo.
+   * Hex value representing characters allowed in URLs. Conventionally, a unique relation that defines the format of
+   * this memo.
    *
    * @return An {@link Optional} of type {@link String} containing the memo type.
    */
   @JsonProperty("MemoType")
-  Optional<String> memoType();
+  public abstract Optional<String> memoType();
+
+  @Check
+  void check() {
+    memoData().ifPresent(
+      memoData -> Preconditions.checkState(isHex(memoData), "MemoData must be a hex-encoded string")
+    );
+    memoFormat().ifPresent(
+      memoFormat -> Preconditions.checkState(isHex(memoFormat), "MemoFormat must be a hex-encoded string")
+    );
+    memoType().ifPresent(
+      memoType -> Preconditions.checkState(isHex(memoType), "MemoType must be a hex-encoded string")
+    );
+  }
+
+  /**
+   * Determines if an input string is hex-encoded.
+   *
+   * @param input A {@link String}.
+   *
+   * @return {@code true} if every character in {@code input} is hex-encoded; {@code false} otherwise.
+   */
+  private static boolean isHex(final String input) {
+    Objects.requireNonNull(input);
+    for (char c : input.toCharArray()) {
+      if (!isHex(c)) {
+        return false;
+      }
+    }
 
+    return true;
+  }
+
+  /**
+   * Determines if an input char is hex-encoded.
+   *
+   * @param input A {@link char}.
+   *
+   * @return {@code true} if {@code input} is hex-encoded; {@code false} otherwise.
+   */
+
+  private static boolean isHex(final char input) {
+    return (input >= '0' && input <= '9') || (input >= 'a' && input <= 'f') || (input >= 'A' && input <= 'F');
+  }
```

Comment:
What happens here when you pass in a `String` with an odd number of characters, which is invalid hex?

Reply:
Confirmed that this works with rippled in an IT.